src/state/selection/logics.ts

```diff
@@ -32,10 +51,75 @@ const storeColorsLogic = createLogic({
             }
             return newAgent;
         });
-        dispatch(receiveAgentNamesAndStates(newUiData));
+        // store color changes in local browser storage
+        const fileKey = getSimulariumFile(getState()).name;
+        localStorage.setItem(fileKey, JSON.stringify(newUiData));
+        // update redux state to store new color changes
+        // and apply them via currentColorSettings
+        dispatch(setUserSelectedUIData(newUiData));
+        dispatch(
+            setCurrentColorSettings({
+                currentColorSettings: ColorSettings.UserSelected,
+            })
+        );
         done();
     },
-    type: SET_COLOR_CHANGES,
+    type: APPLY_USER_COLOR_SELECTION,
 });
 
-export default [storeColorsLogic];
+const clearSessionColorsLogic = createLogic({
+    process(deps: ReduxLogicDeps, dispatch, done) {
+        const { getState } = deps;
+        const fileKey = getSimulariumFile(getState()).name;
+        localStorage.removeItem(fileKey);
+        dispatch(setUserSelectedUIData([]));
+        done();
+    },
+    type: CLEAR_UI_DATA_FROM_BROWSER_AND_STATE,
+});
+
+const applySessionColorsLogic = createLogic({
+    process(deps: ReduxLogicDeps, dispatch, done) {
+        const { getState } = deps;
+
+        const simulariumFile = getSimulariumFile(getState());
+        const defaultUIData = getDefaultUIData(getState());
+        const storedColorChanges = localStorage.getItem(simulariumFile.name);
+        if (storedColorChanges !== "undefined" && storedColorChanges !== null) {
+            const uiData = JSON.parse(storedColorChanges);
+            // if default UI data hasn't been received from the viewer
+            // store these settings but dont apply them until a matching
+            // agent structure is verified
+            if (defaultUIData.length === 0) {
+                dispatch(setUserSelectedUIData(uiData));
+                done();
+                return;
+            }
+            const agentStructuresMatch = compareAgentTrees(
+                uiData,
+                defaultUIData
+            );
+            if (!agentStructuresMatch) {
+                console.warn(
+                    "Agent structures do not match, not applying color settings from browser storage"
+                );
+                done();
+                return;
+            }
+            dispatch(setUserSelectedUIData(uiData));
+            dispatch(
+                setCurrentColorSettings({
+                    currentColorSettings: ColorSettings.UserSelected,
+                })
+            );
+        }
+        done();
```

Comment:
One of two places we check if stored color settings match the current trajectory.

If viewer hasn't provided default data yet, we store browser settings without applying them. The check for whether to apply them will happen in a redux logic when the default arrives.

If default is already in state, we can compare and either apply the stored settings or discard them if the trajectories don't match.